precompile/contracts/rewardmanager/contract_test.go

```diff
@@ -191,112 +184,79 @@ func TestRewardManagerRun(t *testing.T) {
-			Caller: enabledAddr,
-			Input: func(tt *testing.T) []byte {
+			Caller: allowlist.TestEnabledAddr,
+			InputFn: func(t *testing.T) []byte {
 				input, err := PackAllowFeeRecipients()
-				require.NoError(tt, err)
+				require.NoError(t, err)
 
 				return input
-			}(t),
+			},
 			SuppliedGas: AllowFeeRecipientsGasCost - 1,
 			ReadOnly:    false,
 			ExpectedErr: vmerrs.ErrOutOfGas.Error(),
 		},
 		"insufficient gas read current reward address from allowed role": {
-			Caller: enabledAddr,
-			Input: func(tt *testing.T) []byte {
+			Caller: allowlist.TestEnabledAddr,
+			InputFn: func(t *testing.T) []byte {
 				input, err := PackCurrentRewardAddress()
-				require.NoError(tt, err)
+				require.NoError(t, err)
 
 				return input
-			}(t),
+			},
 			SuppliedGas: CurrentRewardAddressGasCost - 1,
 			ReadOnly:    false,
 			ExpectedErr: vmerrs.ErrOutOfGas.Error(),
 		},
 		"insufficient gas are fee recipients allowed from allowed role": {
-			Caller: enabledAddr,
-			Input: func(tt *testing.T) []byte {
+			Caller: allowlist.TestEnabledAddr,
+			InputFn: func(t *testing.T) []byte {
 				input, err := PackAreFeeRecipientsAllowed()
-				require.NoError(tt, err)
+				require.NoError(t, err)
 
 				return input
-			}(t),
+			},
 			SuppliedGas: AreFeeRecipientsAllowedGasCost - 1,
 			ReadOnly:    false,
 			ExpectedErr: vmerrs.ErrOutOfGas.Error(),
 		},
-	} {
-		t.Run(name, func(t *testing.T) {
-			db := rawdb.NewMemoryDatabase()
-			state, err := state.New(common.Hash{}, state.NewDatabase(db), nil)
-			require.NoError(t, err)
-
-			// Set up the state so that each address has the expected permissions at the start.
-			SetRewardManagerAllowListStatus(state, adminAddr, allowlist.AdminRole)
-			SetRewardManagerAllowListStatus(state, enabledAddr, allowlist.EnabledRole)
-			require.Equal(t, allowlist.AdminRole, GetRewardManagerAllowListStatus(state, adminAddr))
-			require.Equal(t, allowlist.EnabledRole, GetRewardManagerAllowListStatus(state, enabledAddr))
-
-			if test.BeforeHook != nil {
-				test.BeforeHook(t, state)
-			}
-
-			blockContext := contract.NewMockBlockContext(testBlockNumber, 0)
-			accesibleState := contract.NewMockAccessibleState(state, blockContext, snow.DefaultContextTest())
-
-			if test.Config != nil {
-				Module.Configure(nil, test.Config, state, blockContext)
-			}
-			ret, remainingGas, err := RewardManagerPrecompile.Run(accesibleState, test.Caller, ContractAddress, test.Input, test.SuppliedGas, test.ReadOnly)
-			if len(test.ExpectedErr) != 0 {
-				require.ErrorContains(t, err, test.ExpectedErr)
-			} else {
-				require.NoError(t, err)
-			}
-
-			require.Equal(t, uint64(0), remainingGas)
-			require.Equal(t, test.ExpectedRes, ret)
-
-			if test.AfterHook != nil {
-				test.AfterHook(t, state)
-			}
-		})
 	}
+
+	allowlist.RunPrecompileWithAllowListTests(t, Module, state.NewTestStateDB, tests)
+	allowlist.RunPrecompileWithAllowListTests(t, Module, state.NewTestStateDB, allowlist.AllowListTests(Module))
```

Comment:
I think we can get rid of one of these lines with the latest changes since they will both run the allow list standard test suite